### ISSUE TITLE
Weak global reference table overflow after a variable time between 10 and 60 minutes

### ISSUE THREAD

**Issue by GeordiRugenbrink** [mapbox/mapbox-gl-native#14599]
When the navigation of the app runs for a certain time, it crashes with the error message shown below. It probably means that you use global references and forget to delete those which creates a memory leak.

```A/project.name: java_vm_ext.cc:662] JNI ERROR (app bug): weak global reference table overflow (max=51200)weak global reference table dump:
    java_vm_ext.cc:662]   Last 10 entries (of 51200):
    java_vm_ext.cc:662]     51199: 0x1fd40000 java.lang.Thread
    java_vm_ext.cc:662]     51198: 0x1fdc0000 java.lang.Thread
    java_vm_ext.cc:662]     51197: 0x1fd80000 java.lang.Thread
    java_vm_ext.cc:662]     51196: 0x1fd00000 java.lang.Thread
    java_vm_ext.cc:662]     51195: 0x1fcc0000 java.lang.Thread
    java_vm_ext.cc:662]     51194: 0x1fc80000 java.lang.Thread
    java_vm_ext.cc:662]     51193: 0x1fc40000 java.lang.Thread
    java_vm_ext.cc:662]     51192: 0x1fc00000 java.lang.Thread
    java_vm_ext.cc:662]     51191: 0x1fb00000 java.lang.Thread
    java_vm_ext.cc:662]     51190: 0x1fb40000 java.lang.Thread
    java_vm_ext.cc:662]   Summary:
    java_vm_ext.cc:662]     34815 of java.lang.Thread (34815 unique instances)
    java_vm_ext.cc:662]     15547 of java.lang.Class (15547 unique instances)
    java_vm_ext.cc:662]       428 of java.lang.DexCache (428 unique instances)
    java_vm_ext.cc:662]       365 of dalvik.system.DexClassLoader (365 unique instances)
    java_vm_ext.cc:662]        21 of android.os.HandlerThread (21 unique instances)
    java_vm_ext.cc:662]         6 of dalvik.system.PathClassLoader (3 unique instances)
    java_vm_ext.cc:662]         4 of com.squareup.picasso.Utils$PicassoThread (4 unique instances)
    java_vm_ext.cc:662]         2 of com.squareup.picasso.Picasso$CleanupThread (2 unique instances)
    java_vm_ext.cc:662]         2 of com.squareup.picasso.Dispatcher$DispatcherThread (2 unique instances)
    java_vm_ext.cc:662]         2 of android.opengl.GLSurfaceView$GLThread (2 unique instances)
    java_vm_ext.cc:662]         2 of android.opengl.GLSurfaceView (2 unique instances)
    java_vm_ext.cc:662]         1 of com.mapbox.services.android.navigation.v5.navigation.RouteProcessorBackgroundThread
    java_vm_ext.cc:662]         1 of okio.AsyncTimeout$Watchdog
    java_vm_ext.cc:662]         1 of java.lang.BootClassLoader
    java_vm_ext.cc:662]         1 of java.lang.ThreadGroup
    java_vm_ext.cc:662]         1 of java.util.TimerThread
    java_vm_ext.cc:662]         1 of android.net.ConnectivityThread
    java_vm_ext.cc:662] 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
    JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
    JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
E/StudioProfiler: JVMTI error: 15(JVMTI_ERROR_THREAD_NOT_ALIVE) 
A/project.name: runtime.cc:558] Runtime aborting...
    runtime.cc:558] Dumping all threads without appropriate locks held: thread list lock
    runtime.cc:558] All threads:
    runtime.cc:558] DALVIK THREADS (214):
    runtime.cc:558] "JDWP Event Helper Thread" prio=5 tid=13 Runnable
    runtime.cc:558]   | group="" sCount=0 dsCount=0 flags=0 obj=0x14740668 self=0x78620f4800
    runtime.cc:558]   | sysTid=13044 nice=0 cgrp=default sched=0/0 handle=0x784a1dd4f0
    runtime.cc:558]   | state=R schedstat=( 56708775406 158986001263 510192 ) utm=3721 stm=1949 core=5 HZ=100
    runtime.cc:558]   | stack=0x784a0e2000-0x784a0e4000 stackSize=1009KB
    runtime.cc:558]   | held mutexes= "abort lock" "JNI weak global reference table lock" "mutator lock"(shared held)
    runtime.cc:558]   native: #00 pc 00000000003c16c8  /system/lib64/libart.so (art::DumpNativeStack(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, int, BacktraceMap*, char const*, art::ArtMethod*, void*, bool)+220)
    runtime.cc:558]   native: #01 pc 000000000048ed34  /system/lib64/libart.so (art::Thread::DumpStack(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, bool, BacktraceMap*, bool) const+352)
    runtime.cc:558]   native: #02 pc 00000000004a88c8  /system/lib64/libart.so (art::DumpCheckpoint::Run(art::Thread*)+844)
    runtime.cc:558]   native: #03 pc 00000000004a16cc  /system/lib64/libart.so (art::ThreadList::RunCheckpoint(art::Closure*, art::Closure*)+484)
    runtime.cc:558]   native: #04 pc 00000000004a0944  /system/lib64/libart.so (art::ThreadList::Dump(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, bool)+500)
    runtime.cc:558]   native: #05 pc 0000000000463ee0  /system/lib64/libart.so (art::Runtime::Abort(char const*)+388)
    runtime.cc:558]   native: #06 pc 0000000000008cd4  /system/lib64/libbase.so (android::base::LogMessage::~LogMessage()+724)
    runtime.cc:558]   native: #07 pc 00000000002e653c  /system/lib64/libart.so (art::JavaVMExt::AddWeakGlobalRef(art::Thread*, art::ObjPtr<art::mirror::Object>)+296)
    runtime.cc:558]   native: #08 pc 00000000003841f0  /system/lib64/libart.so (art::JNI::NewWeakGlobalRef(_JNIEnv*, _jobject*)+612)
    runtime.cc:558]   native: #09 pc 00000000000fdd1c  /system/lib64/libart.so (art::(anonymous namespace)::CheckJNI::NewRef(char const*, _JNIEnv*, _jobject*, art::IndirectRefKind)+760)
    runtime.cc:558]   native: #10 pc 0000000000015130  /system/lib64/libjdwp.so (commonRef_refToID+284)
    runtime.cc:558]   native: #11 pc 00000000000254c4  /system/lib64/libjdwp.so (outStream_writeObjectRef+60)
    runtime.cc:558]   native: #12 pc 0000000000020e2c  /system/lib64/libjdwp.so (commandLoop+1272)
A/project.name: runtime.cc:558]   native: #13 pc 0000000000089694  /system/lib64/libopenjdkjvmti.so (openjdkjvmti::AgentCallback(void*)+1020)
    runtime.cc:558]   native: #14 pc 0000000000081938  /system/lib64/libc.so (__pthread_start(void*)+36)
    runtime.cc:558]   native: #15 pc 0000000000023478  /system/lib64/libc.so (__start_thread+68)
    runtime.cc:558]   (no managed stack frames)
    runtime.cc:558] 
    runtime.cc:558] "main" prio=5 tid=1 Native
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x74eb5b28 self=0x7862014c00
    runtime.cc:558]   | sysTid=13014 nice=-10 cgrp=default sched=0/0 handle=0x78e7ad6548
    runtime.cc:558]   | state=S schedstat=( 272164862721 57439243163 582286 ) utm=23537 stm=3679 core=6 HZ=100
    runtime.cc:558]   | stack=0x7fe57dc000-0x7fe57de000 stackSize=8MB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000001f02c  /system/lib64/libc.so (syscall+28)
    runtime.cc:558]   native: #01 pc 00000000000d724c  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:558]   native: #02 pc 00000000000ff530  /system/lib64/libart.so (art::(anonymous namespace)::CheckJNI::CallMethodV(char const*, _JNIEnv*, _jobject*, _jclass*, _jmethodID*, std::__va_list, art::Primitive::Type, art::InvokeType)+500)
    runtime.cc:558]   native: #03 pc 00000000000ec2cc  /system/lib64/libart.so (art::(anonymous namespace)::CheckJNI::CallObjectMethod(_JNIEnv*, _jobject*, _jmethodID*, ...)+156)
    runtime.cc:558]   native: #04 pc 00000000000f6050  /system/lib64/libandroid_runtime.so (android::NativeDisplayEventReceiver::dispatchVsync(long, int, unsigned int)+48)
    runtime.cc:558]   native: #05 pc 000000000004403c  /system/lib64/libandroidfw.so (android::DisplayEventDispatcher::handleEvent(int, int, void*)+164)
    runtime.cc:558]   native: #06 pc 00000000000141c0  /system/lib64/libutils.so (android::Looper::pollInner(int)+856)
    runtime.cc:558]   native: #07 pc 0000000000013dcc  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+60)
    runtime.cc:558]   native: #08 pc 0000000000121e04  /system/lib64/libandroid_runtime.so (android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long, int)+44)
    runtime.cc:558]   at android.os.MessageQueue.nativePollOnce(Native method)
    runtime.cc:558]   at android.os.MessageQueue.next(MessageQueue.java:326)
    runtime.cc:558]   at android.os.Looper.loop(Looper.java:165)
    runtime.cc:558]   at android.app.ActivityThread.main(ActivityThread.java:-1)
    runtime.cc:558]   at java.lang.reflect.Method.invoke(Native method)
    runtime.cc:558]   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
    runtime.cc:558]   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
    runtime.cc:558] 
    runtime.cc:558] "Jit thread pool worker thread 0" prio=5 tid=2 Native
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14740000 self=0x785b20e000
    runtime.cc:558]   | sysTid=13028 nice=9 cgrp=default sched=0/0 handle=0x785b58d4f0
    runtime.cc:558]   | state=S schedstat=( 8276354005 13459315388 30191 ) utm=587 stm=240 core=7 HZ=100
    runtime.cc:558]   | stack=0x785b48f000-0x785b491000 stackSize=1021KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000001f02c  /system/lib64/libc.so (syscall+28)
    runtime.cc:558]   native: #01 pc 00000000000d724c  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:558]   native: #02 pc 00000000004aa10c  /system/lib64/libart.so (art::ThreadPool::GetTask(art::Thread*)+260)
    runtime.cc:558]   native: #03 pc 00000000004a9674  /system/lib64/libart.so (art::ThreadPoolWorker::Run()+124)
    runtime.cc:558]   native: #04 pc 00000000004a9134  /system/lib64/libart.so (art::ThreadPoolWorker::Callback(void*)+148)
    runtime.cc:558]   native: #05 pc 0000000000081938  /system/lib64/libc.so (__pthread_start(void*)+36)
    runtime.cc:558]   native: #06 pc 0000000000023478  /system/lib64/libc.so (__start_thread+68)
    runtime.cc:558]   (no managed stack frames)
    runtime.cc:558] 
    runtime.cc:558] "Signal Catcher" prio=5 tid=3 WaitingInMainSignalCatcherLoop
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14740088 self=0x7862016400
    runtime.cc:558]   | sysTid=13029 nice=0 cgrp=default sched=0/0 handle=0x785b1ff4f0
    runtime.cc:558]   | state=S schedstat=( 346666 0 1 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:558]   | stack=0x785b104000-0x785b106000 stackSize=1009KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: do_sigtimedwait+0xe0/0x1c0
    runtime.cc:558]   kernel: SyS_rt_sigtimedwait+0xc8/0x118
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000006e23c  /system/lib64/libc.so (__rt_sigtimedwait+8)
    runtime.cc:558]   native: #01 pc 000000000002d8cc  /system/lib64/libc.so (sigwait+64)
    runtime.cc:558]   native: #02 pc 000000000047cba0  /system/lib64/libart.so (art::SignalCatcher::WaitForSignal(art::Thread*, art::SignalSet&)+388)
    runtime.cc:558]   native: #03 pc 000000000047b548  /system/lib64/libart.so (art::SignalCatcher::Run(void*)+272)
    runtime.cc:558]   native: #04 pc 0000000000081938  /system/lib64/libc.so (__pthread_start(void*)+36)
    runtime.cc:558]   native: #05 pc 0000000000023478  /system/lib64/libc.so (__start_thread+68)
    runtime.cc:558]   (no managed stack frames)
    runtime.cc:558] 
    runtime.cc:558] "ADB-JDWP Connection Control Thread" prio=5 tid=4 WaitingInMainDebuggerLoop
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14740110 self=0x785b218800
    runtime.cc:558]   | sysTid=13030 nice=0 cgrp=default sched=0/0 handle=0x784ae394f0
    runtime.cc:558]   | state=S schedstat=( 48861043 7038331 28 ) utm=3 stm=1 core=5 HZ=100
    runtime.cc:558]   | stack=0x784ad3e000-0x784ad40000 stackSize=1009KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: poll_schedule_timeout+0x4c/0x7c
    runtime.cc:558]   kernel: do_sys_poll+0x34c/0x480
    runtime.cc:558]   kernel: SyS_ppoll+0x11c/0x1b4
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000006e17c  /system/lib64/libc.so (__ppoll+8)
    runtime.cc:558]   native: #01 pc 000000000002b5ac  /system/lib64/libc.so (poll+88)
    runtime.cc:558]   native: #02 pc 0000000000006c50  /system/lib64/libadbconnection.so (adbconnection::AdbConnectionState::RunPollLoop(art::Thread*)+836)
    runtime.cc:558]   native: #03 pc 0000000000005218  /system/lib64/libadbconnection.so (adbconnection::CallbackFunction(void*)+1060)
    runtime.cc:558]   native: #04 pc 0000000000081938  /system/lib64/libc.so (__pthread_start(void*)+36)
    runtime.cc:558]   native: #05 pc 0000000000023478  /system/lib64/libc.so (__start_thread+68)
    runtime.cc:558]   (no managed stack frames)
    runtime.cc:558] 
    runtime.cc:558] "HeapTaskDaemon" prio=5 tid=5 WaitingForTaskProcessor
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14740198 self=0x7862090c00
    runtime.cc:558]   | sysTid=13034 nice=4 cgrp=default sched=0/0 handle=0x784aa294f0
    runtime.cc:558]   | state=S schedstat=( 104520295944 10045923027 43030 ) utm=9772 stm=680 core=6 HZ=100
    runtime.cc:558]   | stack=0x784a926000-0x784a928000 stackSize=1041KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   native: #01 pc 00000000000d7678  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+168)
    runtime.cc:558]   native: #02 pc 000000000022cc2c  /system/lib64/libart.so (art::gc::TaskProcessor::GetTask(art::Thread*)+504)
    runtime.cc:558]   native: #03 pc 000000000022d480  /system/lib64/libart.so (art::gc::TaskProcessor::RunAllTasks(art::Thread*)+96)
    runtime.cc:558]   at dalvik.system.VMRuntime.runHeapTasks(Native method)
    runtime.cc:558]   at java.lang.Daemons$HeapTaskDaemon.runInternal(Daemons.java:475)
    runtime.cc:558]   at java.lang.Daemons$Daemon.run(Daemons.java:103)
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558] 
    runtime.cc:558] "FinalizerDaemon" prio=5 tid=6 Waiting
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14740220 self=0x786207b000
    runtime.cc:558]   | sysTid=13032 nice=4 cgrp=default sched=0/0 handle=0x784ac354f0
    runtime.cc:558]   | state=S schedstat=( 2216540203 1524311668 4412 ) utm=208 stm=13 core=7 HZ=100
    runtime.cc:558]   | stack=0x784ab32000-0x784ab34000 stackSize=1041KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000001f02c  /system/lib64/libc.so (syscall+28)
    runtime.cc:558]   native: #01 pc 00000000000d724c  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:558]   native: #02 pc 00000000003bcc6c  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+636)
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   - waiting on <0x04f0ec67> (a java.lang.Object)
    runtime.cc:558]   at java.lang.Object.wait(Object.java:422)
    runtime.cc:558]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:188)
    runtime.cc:558]   - locked <0x04f0ec67> (a java.lang.Object)
    runtime.cc:558]   at java.lang.ref.ReferenceQueue.remove(ReferenceQueue.java:209)
    runtime.cc:558]   at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:232)
    runtime.cc:558]   at java.lang.Daemons$Daemon.run(Daemons.java:103)
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558] 
    runtime.cc:558] "FinalizerWatchdogDaemon" prio=5 tid=7 Sleeping
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x147402a8 self=0x7862090000
    runtime.cc:558]   | sysTid=13033 nice=4 cgrp=default sched=0/0 handle=0x784ab2f4f0
    runtime.cc:558]   | state=S schedstat=( 42428593 209255468 357 ) utm=4 stm=0 core=6 HZ=100
    runtime.cc:558]   | stack=0x784aa2c000-0x784aa2e000 stackSize=1041KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   native: #01 pc 00000000000d7678  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+168)
    runtime.cc:558]   native: #02 pc 00000000003bcc80  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+656)
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
    runtime.cc:558]   at java.lang.Thread.sleep(Native method)
    runtime.cc:558]   - sleeping on <0x01463a14> (a java.lang.Object)
    runtime.cc:558]   at java.lang.Thread.sleep(Thread.java:373)
    runtime.cc:558]   - locked <0x01463a14> (a java.lang.Object)
    runtime.cc:558]   at java.lang.Thread.sleep(Thread.java:314)
    runtime.cc:558]   at java.lang.Daemons$FinalizerWatchdogDaemon.sleepFor(Daemons.java:342)
    runtime.cc:558]   at java.lang.Daemons$FinalizerWatchdogDaemon.waitForFinalization(Daemons.java:364)
    runtime.cc:558]   at java.lang.Daemons$FinalizerWatchdogDaemon.runInternal(Daemons.java:281)
    runtime.cc:558]   at java.lang.Daemons$Daemon.run(Daemons.java:103)
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558] 
    runtime.cc:558] "ReferenceQueueDaemon" prio=5 tid=8 Waiting
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14740338 self=0x7862070400
    runtime.cc:558]   | sysTid=13031 nice=4 cgrp=default sched=0/0 handle=0x784ad3b4f0
    runtime.cc:558]   | state=S schedstat=( 561307551 815542962 1888 ) utm=47 stm=9 core=7 HZ=100
    runtime.cc:558]   | stack=0x784ac38000-0x784ac3a000 stackSize=1041KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000001f02c  /system/lib64/libc.so (syscall+28)
    runtime.cc:558]   native: #01 pc 00000000000d724c  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:558]   native: #02 pc 00000000003bcc6c  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+636)
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   - waiting on <0x001102bd> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
    runtime.cc:558]   at java.lang.Daemons$ReferenceQueueDaemon.runInternal(Daemons.java:178)
    runtime.cc:558]   - locked <0x001102bd> (a java.lang.Class<java.lang.ref.ReferenceQueue>)
    runtime.cc:558]   at java.lang.Daemons$Daemon.run(Daemons.java:103)
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558] 
    runtime.cc:558] "Binder:13014_1" prio=5 tid=9 Native
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x147403c0 self=0x7859627800
    runtime.cc:558]   | sysTid=13035 nice=0 cgrp=default sched=0/0 handle=0x784a8254f0
    runtime.cc:558]   | state=S schedstat=( 1360833721 891678101 2834 ) utm=122 stm=14 core=4 HZ=100
    runtime.cc:558]   | stack=0x784a72a000-0x784a72c000 stackSize=1009KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: binder_thread_read+0x3cc/0x1200
    runtime.cc:558]   kernel: binder_ioctl_write_read.constprop.47+0x1d4/0x310
    runtime.cc:558]   kernel: binder_ioctl+0x204/0x690
    runtime.cc:558]   kernel: do_vfs_ioctl+0x700/0x7e8
    runtime.cc:558]   kernel: SyS_ioctl+0x60/0x88
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000006e148  /system/lib64/libc.so (__ioctl+4)
    runtime.cc:558]   native: #01 pc 00000000000290d0  /system/lib64/libc.so (ioctl+136)
    runtime.cc:558]   native: #02 pc 000000000005ab80  /system/lib64/libbinder.so (android::IPCThreadState::talkWithDriver(bool)+256)
    runtime.cc:558]   native: #03 pc 000000000005ad54  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+24)
    runtime.cc:558]   native: #04 pc 000000000005b4a0  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
    runtime.cc:558]   native: #05 pc 000000000007bb0c  /system/lib64/libbinder.so (android::PoolThread::threadLoop()+24)
    runtime.cc:558]   native: #06 pc 000000000000f9d4  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+280)
    runtime.cc:558]   native: #07 pc 00000000000b6c60  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+140)
    runtime.cc:558]   native: #08 pc 0000000000081938  /system/lib64/libc.so (__pthread_start(void*)+36)
    runtime.cc:558]   native: #09 pc 0000000000023478  /system/lib64/libc.so (__start_thread+68)
    runtime.cc:558]   (no managed stack frames)
    runtime.cc:558] 
    runtime.cc:558] "Binder:13014_2" prio=5 tid=10 Native
A/project.name: runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14740448 self=0x785b22f000
    runtime.cc:558]   | sysTid=13036 nice=0 cgrp=default sched=0/0 handle=0x784a7274f0
    runtime.cc:558]   | state=S schedstat=( 1282819275 1069214389 3282 ) utm=111 stm=17 core=4 HZ=100
    runtime.cc:558]   | stack=0x784a62c000-0x784a62e000 stackSize=1009KB
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #00 pc 000000000001f02c  /system/lib64/libc.so (syscall+28)
    runtime.cc:558]   native: #01 pc 00000000000d724c  /system/lib64/libart.so (art::ConditionVariable::WaitHoldingLocks(art::Thread*)+148)
    runtime.cc:558]   native: #02 pc 00000000000ff530  /system/lib64/libart.so (art::(anonymous namespace)::CheckJNI::CallMethodV(char const*, _JNIEnv*, _jobject*, _jclass*, _jmethodID*, std::__va_list, art::Primitive::Type, art::InvokeType)+500)
    runtime.cc:558]   native: #03 pc 00000000000ec4d4  /system/lib64/libart.so (art::(anonymous namespace)::CheckJNI::CallBooleanMethodV(_JNIEnv*, _jobject*, _jmethodID*, std::__va_list)+92)
    runtime.cc:558]   native: #04 pc 00000000000c6628  /system/lib64/libandroid_runtime.so (_JNIEnv::CallBooleanMethod(_jobject*, _jmethodID*, ...)+120)
    runtime.cc:558]   native: #05 pc 0000000000130688  /system/lib64/libandroid_runtime.so (JavaBBinder::onTransact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+156)
    runtime.cc:558]   native: #06 pc 000000000004fde4  /system/lib64/libbinder.so (android::BBinder::transact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+136)
    runtime.cc:558]   native: #07 pc 000000000005b09c  /system/lib64/libbinder.so (android::IPCThreadState::executeCommand(int)+520)
    runtime.cc:558]   native: #08 pc 000000000005add8  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+156)
    runtime.cc:558]   native: #09 pc 000000000005b4a0  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
    runtime.cc:558]   native: #10 pc 000000000007bb0c  /system/lib64/libbinder.so (android::PoolThread::threadLoop()+24)
    runtime.cc:558]   native: #11 pc 000000000000f9d4  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+280)
    runtime.cc:558]   native: #12 pc 00000000000b6c60  /system/lib64/libandroid_runtime.so (android::AndroidRuntime::javaThreadShell(void*)+140)
    runtime.cc:558]   native: #13 pc 0000000000081938  /system/lib64/libc.so (__pthread_start(void*)+36)
    runtime.cc:558]   native: #14 pc 0000000000023478  /system/lib64/libc.so (__start_thread+68)
    runtime.cc:558]   (no managed stack frames)
    runtime.cc:558] 
    runtime.cc:558] "Binder:13014_3" prio=5 tid=11 Native
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x147404d0 self=0x78597e7000
A/project.name: runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   - waiting on <0x04fda298> (a java.lang.Object)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
    runtime.cc:558]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2059)
    runtime.cc:558]   at java.util.concurrent.ScheduledThreadPoolExecutor$DelayedWorkQueue.take(ScheduledThreadPoolExecutor.java:849)
    runtime.cc:558] "RenderThread" prio=5 tid=42 Native
    runtime.cc:558]   kernel: SyS_epoll_pwait+0xb0/0x124
    runtime.cc:558]   native: #00 pc 000000000006e05c  /system/lib64/libc.so (__epoll_pwait+8)
    runtime.cc:558]   native: #06 pc 0000000000023478  /system/lib64/libc.so (__start_thread+68)
A/project.name: runtime.cc:558]   at okio.AsyncTimeout$Watchdog.run(AsyncTimeout.java:312)
    runtime.cc:558]   native: #02 pc 0000000000013dcc  /system/lib64/libutils.so (android::Looper::pollOnce(int, int*, int*, void**)+60)
    runtime.cc:558]   at android.os.MessageQueue.nativePollOnce(Native method)
    runtime.cc:558]   at android.os.Looper.loop(Looper.java:165)
    runtime.cc:558]   kernel: SyS_epoll_wait+0x288/0x364
    runtime.cc:558]   | stack=0x7844bf6000-0x7844bf8000 stackSize=1041KB
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   | sysTid=13109 nice=10 cgrp=default sched=0/0 handle=0x7844aed4f0
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   at java.util.concurrent.PriorityBlockingQueue.take(PriorityBlockingQueue.java:548)
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   at java.lang.Thread.parkFor$(Thread.java:2137)
    runtime.cc:558]   at sun.misc.Unsafe.park(Unsafe.java:358)
    runtime.cc:558]   at java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2059)
    runtime.cc:558]   native: #03 pc 0000000000438f90  /system/lib64/libhwui.so (android::uirenderer::TaskManager::WorkerThread::threadLoop()+316)
A/project.name: runtime.cc:558]   native: #00 pc 000000000001f02c  /system/lib64/libc.so (syscall+28)
    runtime.cc:558]   native: #02 pc 00000000003bcc6c  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+636)
    runtime.cc:558]   at java.util.concurrent.locks.LockSupport.park(LockSupport.java:190)
    runtime.cc:558]   | sysTid=13197 nice=19 cgrp=default sched=0/0 handle=0x78436034f0
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   native: #02 pc 000000000001469c  /system/lib64/libutils.so (android::Looper::pollAll(int, int*, int*, void**)+288)
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14771f08 self=0x7849d2fc00
    runtime.cc:558]   | state=S schedstat=( 10349254607 15681973435 47607 ) utm=671 stm=363 core=6 HZ=100
    runtime.cc:558]   | stack=0x784330c000-0x784330e000 stackSize=1009KB
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #06 pc 0000000000081938  /system/lib64/libc.so (__pthread_start(void*)+36)
    runtime.cc:558] "OkHttp ConnectionPool" prio=5 tid=71 TimedWaiting
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
A/project.name: runtime.cc:558]   - waiting on <0x0d17004f> (a android.opengl.GLSurfaceView$GLThreadManager)
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
    runtime.cc:558]   at com.mapbox.mapboxsdk.maps.renderer.MapRenderer.onDrawFrame(MapRenderer.java:96)
    runtime.cc:558]   at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1575)
    runtime.cc:558]   native: #00 pc 000000000006e05c  /system/lib64/libc.so (__epoll_pwait+8)
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   native: #01 pc 0000000000013ef8  /system/lib64/libutils.so (android::Looper::pollInner(int)+144)
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   - locked <0x0878bce5> (a java.lang.Object)
    runtime.cc:558] 
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14772650 self=0x7843e5e800
    runtime.cc:558]   | held mutexes=
A/project.name: runtime.cc:558]   | sysTid=13263 nice=0 cgrp=default sched=0/0 handle=0x78396b44f0
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
    runtime.cc:558]   at android.os.MessageQueue.next(MessageQueue.java:326)
    runtime.cc:558] 
    runtime.cc:558]   | state=S schedstat=( 9563125 12533228 70 ) utm=0 stm=0 core=4 HZ=100
    runtime.cc:558] "Binder:13014_5" prio=5 tid=134 Native
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   native: #07 pc 000000000005b09c  /system/lib64/libbinder.so (android::IPCThreadState::executeCommand(int)+520)
    runtime.cc:558]   native: #14 pc 0000000000023478  /system/lib64/libc.so (__start_thread+68)
    runtime.cc:558]   | sysTid=13721 nice=0 cgrp=default sched=0/0 handle=0x783d7ff4f0
A/project.name: runtime.cc:558]   native: #04 pc 000000000005b4a0  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+60)
A/project.name: runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    runtime.cc:558]   - locked <0x0e2d2f6a> (a okhttp3.ConnectionPool)
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   native: #01 pc 00000000000d7678  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+168)
    runtime.cc:558]   at java.lang.Object.wait(Native method)
A/project.name: runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
A/project.name: runtime.cc:558]   at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1091)
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   at java.lang.Thread.parkFor$(Thread.java:2137)
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   | state=S schedstat=( 159375001 99547973 292 ) utm=13 stm=2 core=6 HZ=100
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   - waiting on <0x0f6d140f> (a java.lang.Object)
    runtime.cc:558]   at sun.misc.Unsafe.park(Unsafe.java:358)
A/project.name: runtime.cc:558]   at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:937)
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x1474fdb0 self=0x78249b6000
A/project.name: runtime.cc:558] "OkHttp Dispatcher" prio=5 tid=136 TimedWaiting
    runtime.cc:558]   | state=S schedstat=( 147085787 154982131 405 ) utm=14 stm=0 core=5 HZ=100
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   native: #02 pc 00000000003bcc80  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+656)
A/project.name: runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    runtime.cc:558] "OkHttp Dispatcher" prio=5 tid=143 TimedWaiting
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:558] 
    runtime.cc:558]   | sysTid=26635 nice=0 cgrp=default sched=0/0 handle=0x78262eb4f0
    runtime.cc:558]   native: #01 pc 00000000000d7678  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+168)
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   native: #02 pc 00000000003bcc80  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+656)
    runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
A/project.name: runtime.cc:558]   - waiting on <0x02a2eb91> (a okhttp3.ConnectionPool)
    runtime.cc:558]   | stack=0x7825b97000-0x7825b99000 stackSize=1041KB
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14753c40 self=0x782f98b400
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
    runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
A/project.name: runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:558]   | stack=0x7822efc000-0x7822efe000 stackSize=1041KB
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   at sun.misc.Unsafe.park(Unsafe.java:358)
A/project.name: runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:558]   native: #02 pc 00000000003bcc80  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+656)
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   - locked <0x067e48fc> (a java.lang.Object)
    runtime.cc:558]   - waiting on <0x0cea0b85> (a java.lang.Object)
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558] "OkHttp Dispatcher" prio=5 tid=173 TimedWaiting
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14775c10 self=0x77afd82800
A/project.name: runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   - waiting on <0x08e4f8e8> (a java.lang.Object)
    runtime.cc:558] "OkHttp Dispatcher" prio=5 tid=178 TimedWaiting
    runtime.cc:558]   | sysTid=26741 nice=0 cgrp=default sched=0/0 handle=0x78214e14f0
    runtime.cc:558]   native: #01 pc 00000000000d7678  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+168)
    runtime.cc:558]   at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:230)
    runtime.cc:558]   | sysTid=26743 nice=0 cgrp=default sched=0/0 handle=0x78213db4f0
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   native: #02 pc 00000000003bcc80  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+656)
A/project.name: runtime.cc:558]   at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:230)
    runtime.cc:558] 
    runtime.cc:558] "OkHttp Dispatcher" prio=5 tid=182 TimedWaiting
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    runtime.cc:558] "OkHttp Dispatcher" prio=5 tid=184 TimedWaiting
    runtime.cc:558]   | sysTid=26746 nice=0 cgrp=default sched=0/0 handle=0x782096f4f0
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   at sun.misc.Unsafe.park(Unsafe.java:358)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1091)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:558]   | stack=0x7820660000-0x7820662000 stackSize=1041KB
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558] 
    runtime.cc:558]   | sysTid=26748 nice=0 cgrp=default sched=0/0 handle=0x78208694f0
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
A/project.name: runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558]   | state=S schedstat=( 197425829 206875474 430 ) utm=17 stm=2 core=7 HZ=100
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:461)
    runtime.cc:558]   | stack=0x7820ec0000-0x7820ec2000 stackSize=1041KB
    runtime.cc:558]   at sun.misc.Unsafe.park(Unsafe.java:358)
    runtime.cc:558]   at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:937)
    runtime.cc:558]   | stack=0x7820dba000-0x7820dbc000 stackSize=1041KB
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
    runtime.cc:558] "OkHttp ConnectionPool" prio=5 tid=191 TimedWaiting
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
A/project.name: runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x147768d0 self=0x783d998400
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #01 pc 00000000000d7678  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+168)
    runtime.cc:558]   | held mutexes=
    runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
    runtime.cc:558] "OkHttp ConnectionPool" prio=5 tid=200 TimedWaiting
    runtime.cc:558]   | state=S schedstat=( 1332970 2256718 14 ) utm=0 stm=0 core=5 HZ=100
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558] "OkHttp ConnectionPool" prio=5 tid=202 TimedWaiting
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   - waiting on <0x095634d7> (a okhttp3.ConnectionPool)
    runtime.cc:558] 
    runtime.cc:558]   | state=S schedstat=( 1819062 21091461 24 ) utm=0 stm=0 core=7 HZ=100
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
A/project.name: runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x14776df8 self=0x783b14dc00
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   - waiting on <0x0cc067e2> (a okhttp3.ConnectionPool)
    runtime.cc:558]   at okhttp3.ConnectionPool$1.run(ConnectionPool.java:67)
    runtime.cc:558] "OkHttp ConnectionPool" prio=5 tid=207 TimedWaiting
    runtime.cc:558]   | sysTid=26768 nice=0 cgrp=default sched=0/0 handle=0x78200094f0
    runtime.cc:558]   native: #02 pc 00000000003bcc80  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+656)
    runtime.cc:558]   | group="" sCount=1 dsCount=0 flags=1 obj=0x147770a0 self=0x783d996c00
    runtime.cc:558]   | held mutexes=
A/project.name: runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
    runtime.cc:558]   | state=S schedstat=( 19285364 19723023 22 ) utm=1 stm=0 core=6 HZ=100
    runtime.cc:558] 
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   at java.lang.Object.wait(Native method)
A/project.name: runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   at java.lang.Object.wait(Native method)
    runtime.cc:558] 
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   - waiting on <0x04af0b48> (a java.lang.Object)
    runtime.cc:558]   at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.getTask(ThreadPoolExecutor.java:1091)
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   - locked <0x05a7ede1> (a java.lang.Object)
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   native: #02 pc 00000000003bcc80  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, long, int, bool, art::ThreadState)+656)
    runtime.cc:558]   | state=S schedstat=( 45114112 42578639 224 ) utm=4 stm=0 core=7 HZ=100
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
A/project.name: runtime.cc:558]   native: #01 pc 00000000000d7678  /system/lib64/libart.so (art::ConditionVariable::TimedWait(art::Thread*, long, int)+168)
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: SyS_futex+0x110/0x19c
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
    runtime.cc:558] 
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   | sysTid=26916 nice=19 cgrp=default sched=0/0 handle=0x781e2d14f0
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
    runtime.cc:558]   at java.util.concurrent.SynchronousQueue$TransferStack.awaitFulfill(SynchronousQueue.java:461)
    runtime.cc:558]   at java.util.concurrent.SynchronousQueue$TransferStack.transfer(SynchronousQueue.java:362)
    runtime.cc:558]   kernel: __switch_to+0xac/0xb8
    runtime.cc:558]   kernel: futex_wait+0xe8/0x1ec
A/project.name: runtime.cc:558]   - locked <0x00a36d8c> (a java.lang.Object)
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558]   | stack=0x781e69b000-0x781e69d000 stackSize=1041KB
    runtime.cc:558]   kernel: futex_wait_queue_me+0xcc/0x130
A/project.name: runtime.cc:558]   at java.util.concurrent.SynchronousQueue.poll(SynchronousQueue.java:937)
    runtime.cc:558]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
    runtime.cc:558]   | sysTid=26923 nice=19 cgrp=default sched=0/0 handle=0x781e66c4f0
    runtime.cc:558]   at java.lang.Thread.parkFor$(Thread.java:2137)
    runtime.cc:558]   at java.util.concurrent.locks.LockSupport.parkNanos(LockSupport.java:230)
    runtime.cc:558]   at java.lang.Thread.run(Thread.java:764)
    runtime.cc:558]   | sysTid=26924 nice=19 cgrp=default sched=0/0 handle=0x781e1cb4f0
    runtime.cc:558]   kernel: do_futex+0xdc/0x7e8
    runtime.cc:558]   native: #00 pc 000000000001f030  /system/lib64/libc.so (syscall+32)
    runtime.cc:558]   - locked <0x0c5f6bdb> (a java.lang.Object)
    runtime.cc:558]   kernel: __sys_trace_return+0x0/0x4
    runtime.cc:558]   native: #03 pc 00000000003be728  /system/lib64/libart.so (art::Monitor::Wait(art::Thread*, art::mirror::Object*, long, int, bool, art::ThreadState)+424)
    runtime.cc:558]   - locked <0x0ddfd678> (a java.lang.Object)
    runtime.cc:558]   native: #01 pc 000000000048ed34  /system/lib64/libart.so (art::Thread::DumpStack(std::__1::basic_ostream<char, std::__1::char_traits<char>>&, bool, BacktraceMap*, bool) const+352)
    runtime.cc:558]   native: #12 pc 0000000000081938  /system/lib64/libc.so (__pthread_start(void*)+36)
    runtime.cc:566] JNI ERROR (app bug): weak global reference table overflow (max=51200)weak global reference table dump:
    runtime.cc:566]   Last 10 entries (of 51200):
    runtime.cc:566]     51199: 0x1fd40000 java.lang.Thread
A/project.name: runtime.cc:566]     51191: 0x1fb00000 java.lang.Thread
A/libc: Fatal signal 6 (SIGABRT), code -6 (SI_TKILL) in tid 13044 (JDWP Event Help), pid 13014 (project.name)
```
![image](https://user-images.githubusercontent.com/26089931/57284172-edf01500-70b0-11e9-96f7-31733ca4b216.png)
This is the graph of the memory usage in the profiler, as you can see it slowly goes up till it hits it's peek and crashes.

### Steps to reproduce

 1. Start the navigation of a route
 2. after a random time between 10 minutes and 60 minutes the app crashes

### Expected behavior

Don't crash after a random period of time

### Actual behavior

It crashes after 

### Configuration

**Android versions:** 9
**Device models:** Nokia 7 plus
**Mapbox SDK versions:** 7.3.1

**Comment by LukasPaczos** [comment]
Thanks for reporting the issue @GeordiRugenbrink! Closing as a duplicate of https://github.com/mapbox/mapbox-gl-native/issues/14599.

**Comment by GeordiRugenbrink** [comment]
Uhm that's this issue @LukasPaczos ?

**Comment by LukasPaczos** [comment]
Whoops, thanks @GeordiRugenbrink, I've edited https://github.com/mapbox/mapbox-gl-native/issues/14599#issuecomment-489994940 with the correct link.
